tests/integration/modifiers/scroll-into-view-test.js

```diff
@@ -0,0 +1,89 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, clearRender } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import sinon from 'sinon';
+
+module('Integration | Modifier | scroll-into-view', function (hooks) {
+  setupRenderingTest(hooks);
+  const sandbox = sinon.createSandbox();
+
+  hooks.beforeEach(function () {
+    this.scrollIntoViewSpy = sandbox.spy(Element.prototype, 'scrollIntoView');
+  });
+
+  hooks.afterEach(function () {
+    this.scrollIntoViewSpy = null;
+    sandbox.restore();
+  });
+
+  test('it renders and passes options when shouldScroll is true', async function (assert) {
+    this.options = { test: true };
+
+    await render(
+      hbs`<div {{scroll-into-view options=this.options shouldScroll=true}}></div>`
+    );
+
+    assert.ok(this.scrollIntoViewSpy.called, 'scrollIntoView was called');
+
+    assert.deepEqual(
+      this.scrollIntoViewSpy.args[0][0],
+      this.options,
+      'scrollIntoView was called with correct params'
+    );
+  });
+
+  test('it does not render when shouldScroll is false', async function (assert) {
+    this.options = { test: true };
+
+    await render(
+      hbs`<div {{scroll-into-view options=this.options shouldScroll=false}}></div>`
+    );
+
+    assert.notOk(this.scrollIntoViewSpy.called, 'scrollIntoView was not');
+  });
+
+  test('it renders when shouldScroll resolves to true', async function (assert) {
```

Comment:
Can we restructure this test and the next one to be more like the last test? That is, render the modifier, assert that it didn't do anything, *then* programmatically resolve the promise and do the final assert?

Reply:
Good point. Just added.